Core/Src/main.c

```diff
@@ -87,14 +90,40 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
+  MX_ADC_Init();
   /* USER CODE BEGIN 2 */
-
+  HAL_ADC_Start(&hadc);
+  HAL_TIM_PWM_Start(&htim1, TIM_CHANNEL_1);
   /* USER CODE END 2 */
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
+
+  uint8_t START = 1;
+  uint8_t END = 2;
+  uint16_t MAX_ADC = 1023; // 1023 is the max value that can be stored by a 10 bit value
```

Comment:
Nice, love the constants. you can add a `static const` at the start of the lines to tell the compiler that these won't change.